The Makefile.macos.SEQ patch is largely adapted from Homebrew:
https://github.com/Homebrew/homebrew-core/blob/c6d166f556351ecfade14930021e2217b59279ec/Formula/ipopt.rb#L31

diff --git a/Make.inc/Makefile.inc.generic.SEQ b/Make.inc/Makefile.macos.SEQ
similarity index 96%
copy from Make.inc/Makefile.inc.generic.SEQ
copy to Make.inc/Makefile.macos.SEQ
index 2558f30..dac9334 100644
--- a/Make.inc/Makefile.inc.generic.SEQ
+++ b/Make.inc/Makefile.macos.SEQ
@@ -94,7 +94,7 @@ IORDERINGSC = $(IMETIS) $(IPORD) $(ISCOTCH)
 PLAT    = 
 # Library extension, + C and Fortran "-o" option
 # may be different under Windows
-LIBEXT  = .a
+LIBEXT  = .dylib
 OUTC    = -o 
 OUTF    = -o 
 # RM : remove files
@@ -102,16 +102,15 @@ RM      = /bin/rm -f
 # CC : C compiler
 CC      = cc
 # FC : Fortran 90 compiler
-FC      = f90
+FC      = gfortran
 # FL : Fortran linker
-FL      = f90
+FL      = $(FC)
 # AR : Archive object in a library
 #      keep a space at the end if options have to be separated from lib name
-AR      = ar vr 
+AR      = $(FC) -dynamiclib -undefined dynamic_lookup -Wl,-install_name,@rpath/$(notdir $@) -o 
 # RANLIB : generate index of an archive file
 #   (optionnal use "RANLIB = echo" in case of problem)
-RANLIB  = ranlib
-#RANLIB  = echo
+RANLIB  = echo
 
 # DEFINE HERE YOUR LAPACK LIBRARY
 
@@ -143,8 +142,8 @@ LIBOTHERS = -lpthread
 CDEFS = -DAdd_
 
 #COMPILER OPTIONS
-OPTF    = -O
-OPTC    = -O -I.
+OPTF    = -fPIC -O -fallow-argument-mismatch
+OPTC    = -fPIC -O -I.
 OPTL    = -O
 
 #Sequential:
diff --git a/Makefile b/Makefile
index 3c0f645..55ac6ba 100644
--- a/Makefile
+++ b/Makefile
@@ -59,6 +59,17 @@ $(libdir)/libpord$(PLAT)$(LIBEXT):
 	  cp $(LPORDDIR)/libpord$(LIBEXT) $@; \
 	fi;
 
+install: d
+	if ( test ! -d $(PREFIX)/lib ) ; then mkdir -p $(PREFIX)/lib ; fi
+	if ( test ! -d $(PREFIX)/include ) ; then mkdir -p $(PREFIX)/include ; fi
+	if ( test ! -d $(PREFIX)/include/mumps_seq ) ; then mkdir -p $(PREFIX)/include/mumps_seq ; fi
+	for lib in libseq/*$(LIBEXT) $(libdir)/*$(LIBEXT) ; do \
+	  ginstall -t $(PREFIX)/lib $$lib; \
+	  install_name_tool -id $(PREFIX)/lib/$$(basename $$lib) $(PREFIX)/lib/$$(basename $$lib); \
+	done
+	ginstall -m 644 -t $(PREFIX)/include include/*.h
+	ginstall -m 644 -t $(PREFIX)/include/mumps_seq libseq/mpi.h
+
 clean:
 	(cd src; $(MAKE) clean)
 	(cd examples; $(MAKE) clean)
